src/java/org/apache/cassandra/db/compaction/CompactionController.java

```diff
@@ -76,17 +77,6 @@ public CompactionController(ColumnFamilyStore cfs, Set<SSTableReader> compacting
 
     void maybeRefreshOverlaps()
     {
-        if (NEVER_PURGE_TOMBSTONES)
-        {
-            logger.debug("not refreshing overlaps - running with -Dcassandra.never_purge_tombstones=true");
-            return;
```

Comment:
I think we should still keep the early return here: https://issues.apache.org/jira/browse/CASSANDRA-11831